shop_app.rb

```diff
@@ -0,0 +1,47 @@
+require "bundler/setup"
+require "sinatra/base"
+require "sinatra/reloader"
+
+Dir["./lib/**/*.rb"].each { |ruby_file| require ruby_file }
+require "./db/seeds.rb"
+
+module Shop
+  class App < Sinatra::Base
+    configure :development do
+      register Sinatra::Reloader
+    end
+
+    get "/" do
+      products = FetchProductList.new.call
+      erb :"products/index", locals: { products: products }
+    end
+
+    get "/products/:id" do |id|
+      product = FetchProduct.new.call(id)
+      halt 404 unless product
+      erb :"products/show", locals: { product: product }
+    end
+
+    get "/basket" do
+      basket_items = FetchBasketItems.new.call
+      erb :"basket/show", locals: { items: basket_items }
+    end
+
+    post "/basket" do
+      begin
```

Comment:
Sorry, but I was wrong about that. You need to use `begin - end` with `rescue` inside those block. You can skip them in method definitions, but not in block that is passed to a method (in this case `post`).